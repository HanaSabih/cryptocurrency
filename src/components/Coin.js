import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Coin = ({ id, symbol, price }) => (
  <NavLink className="cardItem d-flex flex-row justify-content-around" to={`/detail/${id}`}>

    <div><h2 className="symbol fw-bold fs-1 py-3">{symbol}</h2></div>
    <div className=" fs-5 py-3">
      <small>{id}</small>
      <p>
        $
        {price}
      </p>
    </div>

  </NavLink>
);

Coin.propTypes = {
  id: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
};

export default Coin;
