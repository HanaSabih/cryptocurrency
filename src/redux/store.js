import {
  combineReducers,
} from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import cryto from './coins/coinsSlice';
import detail from './details/detailsSlice';

const reducer = combineReducers({
  cryto,
  detail,
});

const store = configureStore({
  reducer,
});

export default store;
