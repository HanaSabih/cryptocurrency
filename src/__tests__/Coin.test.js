import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Coin from '../components/Coin';

const cryto = {
  changePercent24Hr: '-2.8770796633988240',
  id: 'bitcoin',
  priceUsd: '29067.8651115798164805',
  symbol: 'BTC',
};

describe('Coin component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Coin
            key={cryto.id}
            id={cryto.id}
            symbol={cryto.symbol}
            price={cryto.priceUsd.substring(0, 8)}
            change={cryto.changePercent24Hr}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
