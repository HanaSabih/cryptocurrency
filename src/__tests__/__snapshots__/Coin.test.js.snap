// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin component component match with the snapshot 1`] = `
<a
  className="cardItem d-flex flex-row justify-content-around"
  href="/detail/bitcoin"
  onClick={[Function]}
>
  <div>
    <h2
      className="symbol fw-bold fs-1 py-3"
    >
      BTC
    </h2>
  </div>
  <div
    className=" fs-5 py-3"
  >
    <small>
      bitcoin
    </small>
    <p>
      $
      29067.86
    </p>
  </div>
</a>
`;
